File: akka-java/src/main/java/org/hahn/spider/actor/ExtractContent.java
package org.hahn.spider.actor;

import akka.actor.UntypedActor;
import org.hahn.spider.message.Extract;
import org.hahn.spider.message.Index;
import org.hahn.spider.message.IndexType;

/**
 * Created by jianghan on 16/5/2.
 */
public class ExtractContent extends UntypedActor {
    @Override
    public void onReceive(Object msg) throws Exception {
        if (msg instanceof Extract) {
            Extract extract = (Extract) msg;
            System.out.println(extract);
            getContext().actorSelection("/user/build").tell(new Index(IndexType.Solr, extract.getUrl(), "title", "content"), getSelf());
            // getSender().tell(new Index(IndexType.Solr, extract.getUrl(), "title", "content"), getSelf());
            // getSender().tell(new Index(IndexType.Titan, extract.getUrl(), "title", "content"), getSelf());
        } else {
            unhandled(msg);
        }
    }
}

File: akka-scala/src/main/scala/hahn/learning/akka/spider/actor/ExtractContent.scala
package hahn.learning.akka.spider.actor

import akka.actor.Actor
import hahn.learning.akka.spider.Message

/**
  * Created by jianghan on 16-5-26.
  */
class ExtractContent extends Actor {

  def receive = {
    case Message.Extract(url, content) => println(s"${url}\t${content.length}")
    case _ => println("Error ExtractContent")
  }

}
